x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/session/Result.java

```diff
@@ -7,8 +7,10 @@
 
 package org.elasticsearch.xpack.esql.session;
 
+import org.elasticsearch.compute.data.Page;
+import org.elasticsearch.compute.operator.DriverProfile;
 import org.elasticsearch.xpack.esql.core.expression.Attribute;
 
 import java.util.List;
 
-public record Result(List<Attribute> columns, List<List<Object>> values) {}
+public record Result(List<Attribute> layout, List<Page> pages, List<DriverProfile> profiles) {}
```

Comment:
super nit: there's also a datatype `Layout` which is not quite just a list of attributes; we sometimes also use `fields` (`schema` would also be appropriate but isn't really used in our code base)